view/src/components/farms.js

```diff
@@ -358,6 +426,344 @@ class Farms extends Component {
+            label="Farm Tags"
+            placeholder="Type or Select a Farm Tag"
+            InputProps={{
+              ...params.InputProps,
+              startAdornment: (
+                <>
+                  <InputAdornment position="start">
+                    <SearchIcon />
+                  </InputAdornment>
+                  {params.InputProps.startAdornment}
+                </>
+              ),
+            }}
+          />
+        )}
+      />
+    );
+  };
+
+  /** Updates filteredData with a new array; used with filters.js queries */
+  updateCards = (newValues) => {
+    if (newValues === []) {
+      return;
+    }
+    this.setState({
+      filteredData: [...newValues],
+    });
+  };
+
+  /** Reset filteredData to the original page data, upon clicking reset */
+  resetCards = () => {
+    this.setState(
+      {
+        filteredData: [...this.state.data],
+        nameQuery: "",
+        locationQuery: "",
+        tagsQuery: [],
+      },
+      this.populateAllFarmTags
+    );
+  };
+
+  /** Renders filtered produce results into Material-UI cards */
+  handleResultsRender = () => {
+    const { classes } = this.props;
+    const { filteredData } = this.state;
+
+    return (
+      <div>
+        <Grid container spacing={2} alignItem="center">
+          {filteredData.map((farm) => (
+            <Grid item xs={12}>
+              <Card className={classes.root} variant="outlined">
+                <CardContent>
+                  <Typography variant="h5" component="h2">
+                    {farm.farmName}
+                  </Typography>
+                  {farm.farmTags.map((tag) => (
+                    <Chip className={classes.chip} label={tag} size="small" />
+                  ))}
+                  <Box
+                    display="flex"
+                    flexDirection="row"
+                    flexWrap="wrap"
+                    p={0}
+                    m={0}
+                  >
+                    <Box p={3}>
+                      <Typography className={classes.pos} color="textSecondary">
+                        Details:
+                      </Typography>
+                      <Typography
+                        variant="body2"
+                        component="p"
+                        className={classes.farmLocation}
+                      >
+                        Location of Farm: {farm.location}
+                      </Typography>
+                    </Box>
+                    <Box p={3}>
+                      <Typography className={classes.pos} color="textSecondary">
+                        Point of Contact:
+                      </Typography>
+                      <Typography variant="body2" component="p">
+                        Name: {farm.contacts[0]["contactName"]}
+                        <br />
+                        Phone: {farm.contacts[0]["contactPhone"]}
+                        <br />
+                        Email: {farm.contacts[0]["contactEmail"]}
```

Comment:
check how Andrew fixed this in a different PR by verifying farms.contacts has length before trying to access the value at [0] (which may not exist and might crash and make you sad)